### PR TITLE
Lazy Loading 

### PR DESCRIPTION
**Pull request by GiannisProkopiou**
The addition of the lazy loading technique to the project, review and annotation process.

### CODE REVIEW

File: label_buddy/projects/helpers.py [hunk 1]
```diff
@@ -581,6 +585,25 @@ def project_statistics(project, user):
...
+def get_audiowaveform_data(audio_file):
+
+    """
+    Return audio waveform data for the task.
+    """
+
+    json_name = audio_file.split(".")[0]
+    json_name = "media/data/" + json_name.split("audio/")[1]
+
+    os.system(f'audiowaveform -i {audio_file[1:]} -o {json_name}.json --pixels-per-second 20 --bits 8')
```
Comment (IoannisSina) [inline label_buddy/projects/helpers.py:598]: Is is possible not to store the file in the database? Maybe you can delete the file after getting the desired information.

File: label_buddy/static/js/annotation_page.js [hunk 2]
```diff
@@ -208,23 +208,36 @@ function fixNumberOfRegions() {
         $(btn).find('#count').text(counter++ + ".")
     }
 }
-//----------------------------------------------------------------------------------------------
 
+function get_loaded_precentage(audio) {
```
Comment (kounelisagis) [inline label_buddy/static/js/annotation_page.js:212]: - no need to multiply by 100
- played variable is never used

File: label_buddy/static/js/annotation_page.js [hunk 3]
```diff
@@ -208,23 +208,36 @@ function fixNumberOfRegions() {
...
+    var buffered = audio.buffered; // returns the buffered portion of the audio
+    var loaded; // the loaded portion of the audio
+    var played; // the played portion of the audio
+
+    if (buffered.length) {
```
Comment (kounelisagis) [inline label_buddy/static/js/annotation_page.js:217]: why do we need this?

File: label_buddy/static/js/annotation_page.js [hunk 4]
```diff
@@ -256,6 +270,17 @@ document.addEventListener('DOMContentLoaded', function() {
         }
     });
 
+    // audioprocess as the audio is playing - calculate the loaded percentage each time
+    wavesurfer.on('audioprocess', function() {
```
Comment (kounelisagis) [inline label_buddy/static/js/annotation_page.js:274]: range should be [0,1] instead of [0,100]

File: label_buddy/static/js/review_page.js [hunk 5]
```diff
@@ -189,21 +189,34 @@ function add_region_to_section(region) {
     let new_region_button = getRegionButton(region);
     $('#regions-div').append(new_region_button);
 }
-//----------------------------------------------------------------------------------------------
 
+function get_loaded_precentage(audio) {
```
Comment (kounelisagis) [inline label_buddy/static/js/review_page.js:193]: same as annotation_page.js

**Review by kounelisagis — CHANGES_REQUESTED** [review]
- Look at the comments.
- Also, identical functions should be declared only once (annotation_page.js - review_page.js). common.js might be a good choice.

File: label_buddy/static/js/annotation_page.js (hunk 3 above)
Comment (GiannisProkopiou) [reply label_buddy/static/js/annotation_page.js:217]: Previous implementation caused an error if the buffer was empty.
Changed it.

**Review by kounelisagis — APPROVED** [review]

**Pull request merged**